packages/ai/src/bootstrap/main.ts

```diff
@@ -27,6 +42,60 @@ const isValidTokenResponse = (data: unknown): data is AIGatewayTokenResponse =>
   )
 }
 
+const isValidProvidersResponse = (data: unknown): data is ProvidersResponse => {
+  return (
+    typeof data === 'object' &&
+    data !== null &&
+    typeof (data as Record<string, unknown>).providers === 'object' &&
+    (data as Record<string, unknown>).providers !== null
+  )
+}
+
+export const fetchAIProviders = async ({ api }: { api: NetlifyAPI }): Promise<AIProviderEnvVar[]> => {
+  try {
+    if (!api.accessToken) {
+      return []
+    }
+
+    const url = `${api.scheme}://${api.host}/api/v1/ai-gateway/providers`
+
+    const response = await fetch(url, {
+      method: 'GET',
+      headers: {
+        Authorization: `Bearer ${api.accessToken}`,
+        'Content-Type': 'application/json',
+      },
+    })
+
+    if (!response.ok) {
+      if (response.status === 404) {
+        return []
+      }
+      throw new Error(`HTTP ${String(response.status)}: ${response.statusText}`)
+    }
+
+    const data: unknown = await response.json()
+
+    if (!isValidProvidersResponse(data)) {
+      throw new Error('Invalid providers response format')
+    }
+
+    const envVars: AIProviderEnvVar[] = []
+
+    for (const [_providerName, provider] of Object.entries(data.providers)) {
```

Comment:
nit: If you're not interested in the keys, you can use `Object.values`.

```suggestion
    for (const provider of Object.values(data.providers)) {
```